src/js/node/net.ts

```diff
@@ -848,18 +974,54 @@ const Socket = (function (InternalSocket) {
       else this.once("finish", this.destroy);
     }
 
+    //TODO: migrate to native
+    _writev(data, callback) {
+      const allBuffers = data.allBuffers;
+      const chunks = data;
+      if (allBuffers) {
+        for (let i = 0; i < data.length; i++) {
+          data[i] = data[i].chunk;
+        }
+      } else {
+        for (let i = 0; i < data.length; i++) {
+          const { chunk, encoding } = data[i];
+          if (typeof chunk === "string") {
+            data[i] = Buffer.from(chunk, encoding);
+          } else {
+            data[i] = chunk;
+          }
+        }
+      }
+      const chunk = Buffer.concat(chunks || []);
```

Comment:
Can you add a check when only one item is passed to data and it's all buffers? This will copy a bunch of data as-is

Reply:
fixed